src/olympia/addons/models.py

```diff
@@ -2440,3 +2452,35 @@ class AddonBrowserMapping(ModelBase):
 
     class Meta:
         unique_together = ('browser', 'extension_id')
+
+
+class DisabledAddonContent(ModelBase):
+    """Link between an addon and the content that was deleted from disk when
+    it was force-disabled.
+
+    That link should be removed if the addon is force-enabled, and the content
+    restored from backup storage."""
+
+    addon = models.OneToOneField(
+        Addon,
+        related_name='content_deleted_on_force_disable',
+        on_delete=models.CASCADE,
+        primary_key=True,
+    )
+    icon_backup_name = models.CharField(
+        max_length=75, default=None, null=True, blank=True
+    )
+    previews = models.ManyToManyField(Preview, through='DeletedPreviewFile')
```

Comment:
This is stricly for convenience - since `DeletedPreviewFile` is the `through` model, this is just for syntaxing sugar (gives you direct access to the previews from the `DisabledAddonContent` model, but having or not doesn't change anything in the database).

I can remove it if you find it more confusing/misleading.

Reply:
I guess that's fine.  Maybe add a comment?

Reply:
I removed it in 37a6e694e30f6df1b321bd0a91de17c546f0d2ac, writing the patch made me realize how useless it was.